documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -834,6 +831,103 @@ public interface IJoystickInputHandler : IInputHandler
 
 `HandleHatMove` must be called when any value of `JoystickState.Hats` changes.
 
+# Input Maps
+
+## Presentation
+Input maps are designed to be easyer to the user to handle actions with more than one input.
+At first view it can looks useless and a bad practice, but this is complemtelly useful for cross-plataform programs, being games an example.
+
+This is an pseudocode example of what the Input Map system solves:
+
+Without inputMaps:
+```cs
+inputContext.Update();
+
+private Dictionary<string, bool> pressedKeys = new();
+
+inputContext.Gamepads.ThumbstickMove += @event => {Move(@event.Value)};
+inputContext.Joysticks.AxisMove += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+With inputMaps:
+```cs
+inputContext.Update();
+
+// This single line get rid of the two in the example above. TWO ISN'T THE LIMIT!
+inputContext.Map.GetAction("Move").Axis += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+## Implementation
+Each InputContext have one reference to an InputMap.
+This is how the InputMap class work:
+```cs
+public partial class InputMap
+{
+    public InputMapAction GetAction(string name);
+    public void AddAction(InputMapAction action, string name);
+    public void RemoveAction(string name);
+    public static InputMap Load(string path);
+    public static void Save(InputMap source, string path);
+    public static string Compile(InputMap source);
+}
+```
+
+`GetAction` method is the main entry point of the system. It allows the use to get an mapped action by it configurated name.
+
+```cs
+public partial class InputMapAction
+{
+    public bool IsPressed { get; }
+    public event Action<AxisInputEvent> Axis;
```

Comment:
How does this work with mapping WASD as a substitute for a thumbstick?

Reply:
It's on the input map syntax example.

If an input doesn't return a vector, you can write

```
- silkVectorClass [

]
```

And the specified inputs inside it, -1 +1, for each axis in the vector class, one input per line.
So like (pseudocode):

```
- Vector2D [
    - A // -x
    - D // +x
    - S // -y
    - W  // +y
]
```

If it's not explained enough in the proposal pls notify

Reply:
Okay, this is not explained in the proposal (at all) particularly your latter code snippet.

Reply:
I will try to improve it more. Also improve the syntax explanation with some other features like ranges and dead radius (this is the name? If it's not sry i will obviously search when i can)

Also some way of returning more customizable values or use modifiers like shift or ctrl